torchtune/modules/low_precision/_state_dict_hooks.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, Tuple
+
+import torch.nn as nn
+from torchao.dtypes.nf4tensor import NF4Tensor
+
+
+def reparametrize_as_bf16_state_dict_post_hook(
+    model: nn.Module,
+    state_dict: Dict[str, Any],
+    *args: Tuple[Any, ...],
+    offload_to_cpu: bool = True,
+    **kwargs: Dict[Any, Any],
```

Comment:
So I'm not sure of the best way to design this sort of API. This API is meant to be a function that's passed into `_register_state_dict_hook`, and the function is invoked with some arguments that we don't use when taking state_dict in nn.Module. Since these are unused arguments but still provided to the function, I need to take them as unused with *args and **kwargs. Though, I want to offer `offload_to_cpu` as a keyword arg with a default, so I can't put it before *args. So this is what I was able to come up with, not sure if there's somethign better.